### PR TITLE
Mlflow example

### PR DESCRIPTION
**Pull request by kshitij12345**
# Description

Include a summary of the changes and the related issue.

__Related to:__ <ClickUp/JIRA task name>

__Any expected test failures?__ macOS Python 3.7 (latest mlflow version is not available)

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [x] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Precommit installed and run before pushing changes
- [x] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [ ] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS: Ubuntu
- Python version: 3.10
- Neptune version: 1.3.1
- Affected libraries with version: tensorflow 2.12, mlflow 2.4.1

### CODE REVIEW

**Comment by kshitij12345** [comment]
@SiddhantSadangi can you have a code review on the current code. (Failures are via integration and will reach out to relevant folks)

Thanks!

**Comment by kshitij12345** [comment]
Gentle Ping @SiddhantSadangi

**Comment by SiddhantSadangi** [comment]
It's on my backlog... will get to it later today/early tomorrow

**Comment by SiddhantSadangi** [comment]
Couple of observations:
1. Do we need hardware monitoring for the Neptune run? It will capture the hardware consumption during the copy operation, not the model training
https://github.com/neptune-ai/neptune-mlflow/blob/master/src/neptune_mlflow_exporter/impl/orchestrator.py#L51

2. I am getting the below error on running `neptune mlflow` locally:
```
Export MLFlow runs to Neptune...
Traceback (most recent call last):
  File "C:\Users\siddh\miniconda3\envs\py38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\siddh\miniconda3\envs\py38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\siddh\miniconda3\envs\py38\Scripts\neptune.exe\__main__.py", line 4, in <module>
  File "C:\Users\siddh\miniconda3\envs\py38\lib\site-packages\neptune\cli\__main__.py", line 39, in <module>
    main.add_command(entry_point.load(), name)
  File "C:\Users\siddh\miniconda3\envs\py38\lib\site-packages\pkg_resources\__init__.py", line 2516, in load
    self.require(*args, **kwargs)
  File "C:\Users\siddh\miniconda3\envs\py38\lib\site-packages\pkg_resources\__init__.py", line 2539, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "C:\Users\siddh\miniconda3\envs\py38\lib\site-packages\pkg_resources\__init__.py", line 827, in resolve
    dist = self._resolve_dist(
  File "C:\Users\siddh\miniconda3\envs\py38\lib\site-packages\pkg_resources\__init__.py", line 868, in _resolve_dist
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'neptune-client>=0.9.0' distribution was not found and is required by the application
```

```
❯ pip list|grep neptune
kedro-neptune                0.1.6
neptune                      1.3.1
neptune-fastai               1.0.0
neptune-lightgbm             1.0.0
neptune-mlflow               1.0.0rc1
neptune-notebooks            0.9.3
neptune-optuna               1.1.0
neptune-prophet              1.0.0
neptune-pytorch              0.2.0
neptune-sacred               1.0.0
neptune-sklearn              2.1.0
neptune-tensorflow-keras     2.1.1
neptune-xgboost              1.1.0
```

Looks like it is searching for `neptune-client`. But tests run fine with `neptune` so no idea what is happening here

File: integrations-and-supported-tools/mlflow/scripts/run_examples.sh
Comment (normandy7) [inline integrations-and-supported-tools/mlflow/scripts/run_examples.sh:12]:
```suggestion
echo "Export MLflow runs to Neptune..."
```

File: README.md
Comment (normandy7) [inline README.md:108]: Looks like the Neptune example was copied over from the previous row.

**Review by normandy7 — COMMENTED** [review]
So the idea with this example is to just log some stuff to MLflow and then export that to Neptune?

File: README.md
Comment (SiddhantSadangi) [reply README.md:108]: Showcase run isn't created yet. Will do once the integration is updated

**Comment by SiddhantSadangi** [comment]
> So the idea with this example is to just log some stuff to MLflow and then export that to Neptune?

yup. the only utility i see for this currently is to have the integration tested regularly through the examples test framework.
But once we release the mlflow sync integration, that would add more value here.

Till then, can we add the exporter to the migration tools section of the docs?

File: README.md
Comment (kshitij12345) [reply README.md:108]: Have updated the integration to address the required changes, will update the requirements.txt to reflect that. Post that we can add a showcase-run.

**Comment by kshitij12345** [comment]
@SiddhantSadangi have updated the run (and tagged showcase run), if everything looks good, will make a 1.0 release

PTAL, Thanks!

**Comment by SiddhantSadangi** [comment]
LGTM! 🎉 

Will approve and merge after unpinning the requirements post-release.

Thanks!

**Comment by normandy7** [comment]
> > So the idea with this example is to just log some stuff to MLflow and then export that to Neptune?
> 
> yup. the only utility i see for this currently is to have the integration tested regularly through the examples test framework. But once we release the mlflow sync integration, that would add more value here.
> 
> Till then, can we add the exporter to the migration tools section of the docs?

Are you suggesting we should already encourage people to install the pre-release version in order to use the exporter? I'd like to get some more input from others if that's the case. My impression is we agreed to wait until everything is ready and then release the complete integration with both the exporter and plugin.

**Comment by SiddhantSadangi** [comment]
> Are you suggesting we should already encourage people to install the pre-release version in order to use the exporter?

Nope. What I meant was, since the initial release (v1.0) would only have the exporter and not the sync ability (that lets to log to both MLflow and Neptune in parallel), it would make more sense to have this in the "Migration Tools" sub-category

**Review by normandy7 — DISMISSED** [review]

**Comment by normandy7** [comment]
> > Are you suggesting we should already encourage people to install the pre-release version in order to use the exporter?
> 
> Nope. What I meant was, since the initial release (v1.0) would only have the exporter and not the sync ability (that lets to log to both MLflow and Neptune in parallel), it would make more sense to have this in the "Migration Tools" sub-category

I understand, I just wasn't aware the scope of the 1.0 release had changed.

File: integrations-and-supported-tools/mlflow/scripts/run_examples.sh
Comment (normandy7) [inline integrations-and-supported-tools/mlflow/scripts/run_examples.sh:13]: How come `sync` is not needed here?

Comment (SiddhantSadangi) [reply integrations-and-supported-tools/mlflow/scripts/run_examples.sh:13]: This is just the exporter to export historical MLflow runs to Neptune. The `sync` feature is still under development

Comment (SiddhantSadangi) [reply integrations-and-supported-tools/mlflow/scripts/run_examples.sh:13]: Also, `sync` would most probably be a python client option, not CLI

Comment (normandy7) [reply integrations-and-supported-tools/mlflow/scripts/run_examples.sh:13]: I'm quite confused. I thought it was part of the exporter command: https://github.com/neptune-ai/neptune-mlflow/blob/6d889e1b2a875e443c7b28f170b01fd8ab85492e/src/neptune_mlflow_exporter/__init__.py#L41C4-L41C4

Comment (SiddhantSadangi) [reply integrations-and-supported-tools/mlflow/scripts/run_examples.sh:13]: Nope, thats the method called by `neptune mlflow` command. The `click.options` are additional flags you can add.

```bash
❯ neptune mlflow --help
Usage: neptune mlflow [OPTIONS]

  Exports MLflow runs to neptune.ai.

  Args:     project: name of the Neptune project where your MLflow runs will
  go.         If not provided, NEPTUNE_PROJECT env variable will be used.
  api_token: your Neptune API token.         If not provided,
  NEPTUNE_API_TOKEN env variable will be used.     mlflow_tracking_uri: your
  MLflow tracking URI.         If not provided, it is left to the MLflow
  client to resolve it.     exclude_artifacts: whether to skip uploading
  artifacts to Neptune.     max_artifact_size: max size of the artifact to be
  uploaded to Neptune.         Unit is in MB.         For directories this
  will be treated as the max size of the entire directory.

Options:
  -p, --project TEXT              Neptune project name
  -a, --api-token TEXT            Your Neptune API Key
  -u, --mlflow-tracking-uri TEXT  Your MLflow tracking URI
  -e, --exclude-artifacts BOOLEAN
                                  Specifies whether to also include artifacts
                                  in the upload
  -m, --max-artifact-size INTEGER
                                  Maximum size uploaded to Neptune, unit is in
                                  MB
  --help                          Show this message and exit.
  ```

Comment (normandy7) [reply integrations-and-supported-tools/mlflow/scripts/run_examples.sh:13]: Ahh, got it. Thanks for explaining!

**Comment by kshitij12345** [comment]
FYI, 1.0 is released

**Review by SiddhantSadangi — APPROVED** [review]

**Pull request merged**
